lib/file_transfer_agent/gcs_util.js

```diff
@@ -137,7 +137,7 @@ function GCSUtil(httpclient, filestream) {
         let matDescKey;
 
         try {
-          if (accessToken) {
+          if (accessToken && !process.env.GOOGLE_APPLICATION_CREDENTIALS) {
```

Comment:
1. As stated above this workaround will work only for the cases when the Google authentication is done with env variables. 
And with the other two methods of authentication this fix won't work.

2. Other than that, if the auth is handled, when the axios.head() request is sent, an error is thrown, which looks like this: `AxiosError: Request failed with status code 404.`
Before the latest updates(I have checked with 1.6.3) it seems that the same error is handled [by checking](https://github.com/snowflakedb/snowflake-connector-nodejs/blob/v1.6.3/lib/file_transfer_agent/gcs_util.js#L168C11-L168C42) if the status is 404 and not throwing the error.
But with this new version the `errCode` is [calculated](https://github.com/snowflakedb/snowflake-connector-nodejs/pull/754/files#diff-284c7e4be78ca4c5a17e537e54884c43b1e7f837ec9877eb415078b138ee021eR180) in a way that the same check does not work, because in the error thrown by Axios `err['code'] = ERR_BAD_REQUEST` and the error should be handled with `err.response.status` in addiction to rest of the status checks. 


cc: @pharturs

Reply:
It was fixed in pr #764